### DISCUSSION TITLE
Reverberation Concerns with Fluidsynth 2.2.8

### DISCUSSION THREAD

**Discussion by JOHNjORGAN** [FluidSynth/fluidsynth#1139] [General]
jOrgan, an Open Source Virtual Pipe Organ program, makes extensive use of Fluidsynth as our sound engine of choice. Reverberation is very important to us. Whilst we are pleased with the improved quality of the reverb sound in Fluidsynth 2 compared with FS 1, it is marred, at least for our use, by a cyclic modulation at a rate of about once each second. It does seem to be a modulation of amplitude rather than pitch.

I am sending this in the hope that developers can address this matter. (Sorry -  the Title should be version 2.2.8)

Best wishes,
JOHNjORGAN

**Comment by derselbst** [comment]
Yes, there is a vibrato effect. This is intentional, see https://github.com/FluidSynth/fluidsynth/blob/e64ad841cb4b6afdc66a0231835b16638108c727/src/rvoice/fluid_rev.c#L256-L273

I don't like it too much either. To address this I was thinking about making the reverb engine configurable. I.e. re-introduce the old freeverb engine and an additional Lexicon-like reverbator. To do so, I would like to rewrite the current reverb implementation to a C++ abstract class hierarchy, to make it easy implementing more reverb engines. But so far I had no time to do so.

**Reply by JOHNjORGAN** [reply]
Thank you very much for your reply, which explains it all. I have conveyed this information to the jOrgan Mailing List. It is clear that the type of music and the reverb settings make a big difference to how intrusive that modulation is. There is no doubt that the use of this technique has produced a better sound for most users. There is much less "selective" ringing.

**Comment by moonskin** [comment]
I have also worked with jOrgan but mostly with Miditzer.

To say this "effect" is "intentional" is, to me, saying that there is a bug here that is unresolved.  Any reverberator that adds an unwanted modulation to a steady tone is faulty.  This needs to be fixed IMO.  I would be inclined to report it as a bug if I could see how.

Regards
Graham

**Reply by derselbst** [reply]
> Any reverberator that adds an unwanted modulation to a steady tone is faulty. This needs to be fixed IMO.

I welcome your PR.

**Reply by jimhen3ry** [reply]
I have submitted this as Issue #1215 on behalf of moonskin and JOHNjORGAN.
